### ISSUE TITLE
print the HTTP status reason phrase when a response returns 4xx,5xx code

### ISSUE THREAD

**Issue by peterlynch** [npm/npm#11594]
The HTTP spec allows augmenting the Status line Reason Phrase from the typical messages.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html#sec6.1.1

A registry may want to change the reason phrase to include more meaningful text that can help an end user understand why they are receiving a certain response code.

For example, instead of 

403 Forbidden

a registry may want to reveal

403 Forbidden blocked

or 

403 Forbidden quarantined
### Expected

Have the npm client render such reason phrases from the status line at the default logging levels to help the user understand problems when encountered.

So for example instead of 

```
npm info retry fetch attempt 2 at 16:41:13
npm info attempt registry request try #1 at 16:41:13
npm http fetch GET http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm http fetch 403 http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm ERR! fetch failed http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 403
```

Include the reason phrase if provided by the  server:

```
npm info retry fetch attempt 2 at 16:41:13
npm info attempt registry request try #1 at 16:41:13
npm http fetch GET http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm http fetch 403 http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm ERR! fetch failed http://127.0.0.1:8081/registry/uglify-js/-/uglify-js-2.2.5.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 403 Quarantined
```

Basically form the status message like [the code here does](https://github.com/npm/npm/blob/master/node_modules/npm-registry-client/lib/request.js#L252).

Instead of [how this code does it](https://github.com/npm/npm/blob/master/node_modules/npm-registry-client/lib/fetch.js#L51)

Ideally all types of 4xx,5xx reason phrases should be handled this way including metadata retrievals and actual package files.
